providers/google.go

```diff
@@ -155,31 +160,49 @@ func (p *GoogleProvider) Redeem(ctx context.Context, redirectURL, code string) (
 
 	c, err := claimsFromIDToken(jsonResponse.IDToken)
 	if err != nil {
-		return
+		return nil, err
 	}
 
 	created := time.Now()
 	expires := time.Now().Add(time.Duration(jsonResponse.ExpiresIn) * time.Second).Truncate(time.Second)
-	s = &sessions.SessionState{
+
+	return &sessions.SessionState{
 		AccessToken:  jsonResponse.AccessToken,
 		IDToken:      jsonResponse.IDToken,
 		CreatedAt:    &created,
 		ExpiresOn:    &expires,
 		RefreshToken: jsonResponse.RefreshToken,
 		Email:        c.Email,
 		User:         c.Subject,
-	}
-	return
+	}, nil
+}
+
+// EnrichSessionState checks the listed Google Groups configured and adds any
+// that the user is a member of to session.Groups.
+func (p *GoogleProvider) EnrichSessionState(ctx context.Context, s *sessions.SessionState) error {
+	p.groupValidator(s)
```

Comment:
Should we not be checking the result of this call?

Reply:
The `bool` output of that method was from the legacy `authorization` check (which is now done in `Authorize` based on the contents on `session.Groups`.

The bool output looks to only be used in the `RefreshSession` refreshing of the groups still.

Reply:
I'm really not a fan of this `groupValidator` -- I just wanted to get it refactored to fit the new `Authorize` method with all existing logic flows persisting so we could fully deprecate `ValidateGroup(string)` on the provider interface that only this users (that `Authorize` fulfills).

Should we make another Issue/PR to give the google provider a much better refactor besides the few areas it was impacted by the move to `Authorize`?

Reply:
If we aren't checking it, how about adding a comment to explain why this is called but not checked?